wasm/jsapi/instance/constructor-caching.any.js

```diff
@@ -0,0 +1,59 @@
+// META: global=jsshell
+// META: script=/wasm/jsapi/wasm-constants.js
+// META: script=/wasm/jsapi/wasm-module-builder.js
+
+function getExports() {
+  const builder = new WasmModuleBuilder();
+  builder
+    .addFunction("fn", kSig_v_d)
+    .addBody([])
+    .exportFunc();
+
+  builder.setTableLength(1);
+  builder.addExportOfKind("table", kExternalTable, 0);
+
+  builder.addGlobal(kWasmI32, false)
+    .exportAs("global")
+    .init = 7;
+
+  builder.addMemory(4, 8, true);
+
+  const buffer = builder.toBuffer();
+  const module = new WebAssembly.Module(buffer);
+  const instance = new WebAssembly.Instance(module);
+  return instance.exports;
+}
+
+test(() => {
+  const exports = getExports();
+
+  const builder = new WasmModuleBuilder();
+  const functionIndex = builder.addImport("module", "imported", kSig_v_d);
+  builder.addExport("exportedFunction", functionIndex);
+
+  const globalIndex = builder.addImportedGlobal("module", "global", kWasmI32);
+  builder.addExportOfKind("exportedGlobal", kExternalGlobal, globalIndex);
+
+  builder.addImportedMemory("module", "memory", 4);
+  builder.exportMemoryAs("exportedMemory");
+
+  const tableIndex = builder.addImportedTable("module", "table", 1);
+  builder.addExportOfKind("exportedTable", kExternalTable, tableIndex);
+
+  const buffer = builder.toBuffer();
+
+  const module = new WebAssembly.Module(buffer);
+  const instance = new WebAssembly.Instance(module, {
+    "module": {
+      "imported": exports.fn,
+      "global": exports.global,
+      "memory": exports.memory,
+      "table": exports.table,
```

Comment:
You could do `{ module: exports }` as well, but might make it harder to understand

Reply:
I think it would be too opaque, yeah.